shesha-core/src/Shesha.Core/Migrations/M20241203150000.cs

```diff
@@ -0,0 +1,338 @@
+                                                       'Core_Notifications')
+                                   OR EXISTS (
+                                      SELECT 1 
+                                      FROM information_schema.constraint_column_usage ccu
+                                      WHERE ccu.constraint_name = tc.constraint_name
+                                        AND ccu.table_name IN ('Core_NotificationMessages', 
+                                                             'Core_NotificationTemplates', 
+                                                             'Core_Notifications')
+                                   )))
+                        LOOP
+                            EXECUTE 'ALTER TABLE ' || quote_ident(r.table_schema) || '.' || quote_ident(r.table_name) 
+                                || ' DROP CONSTRAINT ' || quote_ident(r.constraint_name);
+                        END LOOP;
+
+                        -- Drop primary keys
+                        FOR r IN (SELECT tc.table_schema, 
+                                       tc.constraint_name, 
+                                       tc.table_name
+                                FROM information_schema.table_constraints tc
+                                WHERE tc.constraint_type = 'PRIMARY KEY'
+                                  AND tc.table_name IN ('Core_NotificationMessages', 
+                                                      'Core_NotificationTemplates', 
+                                                      'Core_Notifications'))
+                        LOOP
+                            EXECUTE 'ALTER TABLE ' || quote_ident(r.table_schema) || '.' || quote_ident(r.table_name) 
+                                || ' DROP CONSTRAINT ' || quote_ident(r.constraint_name);
+                        END LOOP;
+
+                        -- Drop default constraints and check constraints
+                        FOR r IN (SELECT tc.table_schema, 
+                                       tc.constraint_name, 
+                                       tc.table_name
+                                FROM information_schema.table_constraints tc
+                                WHERE tc.constraint_type IN ('CHECK', 'DEFAULT')
+                                  AND tc.table_name IN ('Core_NotificationMessages', 
+                                                      'Core_NotificationTemplates', 
+                                                      'Core_Notifications'))
+                        LOOP
+                            EXECUTE 'ALTER TABLE ' || quote_ident(r.table_schema) || '.' || quote_ident(r.table_name) 
+                                || ' DROP CONSTRAINT IF EXISTS ' || quote_ident(r.constraint_name);
+                        END LOOP;
+                    END $$;
+                ");
+
+                // For PostgreSQL, we also need to drop default values set using ALTER COLUMN
+                Execute.Sql(@"
```

Comment:
Migration fails here on MS Sql because this statement is executed without type of DBMS. Note: you needn't any complex sql for dropping/altering of constraints/indexes. Their names must be identical in any DB of the save version.